plugin/src/main/java/com/denizenscript/denizen/objects/properties/entity/EntityVisible.java

```diff
@@ -29,86 +27,90 @@ public static EntityVisible getFrom(ObjectTag entity) {
         }
         else {
-            return ((ItemFrame) entity.getBukkitEntity()).isVisible();
+            return !entity.getLivingEntity().isInvisible();
         }
     }
 
     @Override
     public String getPropertyString() {
-        if (!isVisible()) {
-            return "false";
-        }
-        return null;
+        return isVisible() ? null : "false";
     }
 
     @Override
     public String getPropertyId() {
         return "visible";
     }
 
-    @Override
-    public ObjectTag getObjectAttribute(Attribute attribute) {
-
-        if (attribute == null) {
-            return null;
-        }
+    public static void registerTags() {
 
         // <--[tag]
         // @attribute <EntityTag.visible>
         // @returns ElementTag(Boolean)
         // @mechanism EntityTag.visible
         // @group attributes
         // @description
-        // Returns whether the armor stand or item frame is visible.
+        // Returns whether the entity is visible.
+        // Supports armor stands, item frames, and living entities.
         // -->
-        if (attribute.startsWith("visible")) {
-            return new ElementTag(isVisible()).getObjectAttribute(attribute.fulfill(1));
-        }
-
-        return null;
-    }
-
-    @Override
-    public void adjust(Mechanism mechanism) {
+        PropertyParser.registerTag(EntityVisible.class, ElementTag.class, "visible", (attribute, object) -> {
+            return new ElementTag(object.isVisible());
+        });
 
         // <--[mechanism]
         // @object EntityTag
         // @name visible
         // @input ElementTag(Boolean)
         // @description
-        // Sets whether the armor stand or item frame is visible.
+        // Sets whether the entity is visible.
+        // Supports armor stands, item frames, and living entities.
         // @tags
         // <EntityTag.visible>
         // -->
-        if (mechanism.matches("visible") && mechanism.requireBoolean()) {
-            if (Depends.citizens != null && entity.isLivingEntity() && CitizensAPI.getNPCRegistry().isNPC(entity.getLivingEntity())) {
-                InvisibleTrait.setInvisible(entity.getLivingEntity(), CitizensAPI.getNPCRegistry().getNPC(entity.getBukkitEntity()), !mechanism.getValue().asBoolean());
-            }
-            else if (entity.getBukkitEntity() instanceof ArmorStand) {
-                ((ArmorStand) entity.getBukkitEntity()).setVisible(mechanism.getValue().asBoolean());
-            }
-            else {
-                ((ItemFrame) entity.getBukkitEntity()).setVisible(mechanism.getValue().asBoolean());
+        PropertyParser.registerMechanism(EntityVisible.class, ElementTag.class, "visible", (object, mechanism, input) -> {
+            if (mechanism.requireBoolean()) {
+                if (object.entity.isCitizensNPC()) {
+                    InvisibleTrait.setInvisible(object.entity.getLivingEntity(), object.entity.getDenizenNPC().getCitizen(), !mechanism.getValue().asBoolean());

```

Comment:
This seems to be ignoring the `input`

Reply:
My bad, was moving the old mech code over and missed that - updated to properly use `input`